docs/source/production.rst

```diff
@@ -85,3 +85,32 @@ such as:
   lines). The default is `origin/master`.
 * Configuring `diff-quality` to return an error code if the quality is too low
 * Troubleshooting
+
+Using `pre-commit`_
+^^^^^^^^^^^^^^^^^^^
+
+SQLFluff comes with two `pre-commit`_ hooks:
+
+* sqlfluff-lint: returns linting errors.
+* sqlfluff-fix: attempts to fix rule violations.
+
+Running with `pre-commit`_, you get the benefit of
+only linting/fixing the files that changed.
+
+Your `.pre-commit-config.yaml` should look like this:
+
+.. code-block:: yaml
+
+  repos:
+  - repo: https://github.com/sqlfluff/sqlfluff
+    rev: v0.4.0
+    hooks:
+      - id: sqlfluff-lint
+        # For dbt projects, this installs the dbt "extras":
```

Comment:
Explain when you need this or not? Or link to somewhere else in the docs that does

Reply:
Added a note https://github.com/sqlfluff/sqlfluff/pull/576/files#diff-4fc96f7f0c2ce176c6648464700d7924432d594701d88e49facbe97ec6d45fd9R127